js/src/sheet.ts

```diff
@@ -212,8 +211,17 @@ let SheetModel = widgets.DOMWidgetModel.extend({
             }
             data[i] = row;
         }
-        this.set('data', data);
-        this.save_changes();
+        this.set_data(data);
+    },
+    set_data: function(new_value) {
+        this.set('data', new_value);
+
+        // Workaround for:
+        // - sending the changes to Python
+        // - triggering the change event
+        // This workaround is faster than making a deep copy of data
+        this.sync("update", this);
+        this.trigger("change:data");
```

Comment:
This is where the magic happens. Instead of making deep copies of the data whenever I want to update them, I use the `set_data` method which will forcibly trigger events and JavaScript -> Python updates.